### PR TITLE
Post-merge code review - updates

### PR DESCRIPTION
**Pull request by lknix**
@MoMannn and @xpepermint can you take a look? Please go commit by commit - I've made them as atomic as possible. Let me know if there are any you don't like or don't make sense.

### CODE REVIEW

File: contracts/tokens/ERC721implementation.sol [hunk 1]
```diff
@@ -230,6 +230,7 @@ contract ERC721implementation is Ownable, ERC721, ERC165implementation {
     external
   {
     require(_operator != address(0));
+    require(ownerToNFTokenCount[msg.sender] > 0);
```
Comment (MoMannn) [inline contracts/tokens/ERC721implementation.sol]: Why is this necessary? It costs additional gas for token owners and does no harm if someone else uses it except that they have to pay gas.. And there could as well be a use case where someone gives operator rights even before he gets a token..

File: contracts/tokens/ERC721EnumerableImplementation.sol [hunk 2]
```diff
@@ -14,11 +14,6 @@ contract ERC721EnumerableImplementation is ERC721implementation {
    */
   uint256[] internal tokens;
 
-  /*
```
Comment (MoMannn) [inline contracts/tokens/ERC721EnumerableImplementation.sol]: We decided to implement a basic _burn function so this mappings will need to stay.

File: contracts/tokens/ERC721implementation.sol (hunk 1 above)
Comment (lknix) [reply contracts/tokens/ERC721implementation.sol]: @MoMannn everything is in the commit message. It's a proposal. wdyt?

File: contracts/tokens/ERC721EnumerableImplementation.sol (hunk 2 above)
Comment (lknix) [reply contracts/tokens/ERC721EnumerableImplementation.sol]: We should put it back when we re-implement burn.

File: contracts/tokens/ERC721implementation.sol (hunk 1 above)
Comment (MoMannn) [reply contracts/tokens/ERC721implementation.sol]: I would leave it as it is.. Like I mentioned someone could give operator rights before he owns tokens and even otherwise it does no damage but it uses gas..

File: contracts/tokens/ERC721EnumerableImplementation.sol (hunk 2 above)
Comment (MoMannn) [reply contracts/tokens/ERC721EnumerableImplementation.sol]: Ok, #13 has burn implemented.

File: test/tokens/ERC721.test.js [hunk 3]
```diff
@@ -40,8 +40,8 @@ contract('ERC721', (accounts) => {
     await assertRevert(nftoken.mint('0', id3));
   });
 
-  it('throws when trying to mint NFToken from non owner ot authorized address', async () => {
-    await assertRevert(nftoken.mint('0', id3, { from: accounts[1] }));
+  it.skip('throws when trying to mint NFToken from non-owner', async () => {
+    await assertRevert(nftoken.mint(accounts[1], id3, { from: accounts[1] }));
```
Comment (lknix) [inline test/tokens/ERC721.test.js]: @MoMannn we need to decide what to do with this. Maybe we should just remove it since we don't have any permission modifiers on `_mint`?

Comment (MoMannn) [reply test/tokens/ERC721.test.js]: We should remove it.
